libs/cardano-ledger-api/src/Cardano/Ledger/Api/Era.hs

```diff
@@ -55,26 +65,440 @@ module Cardano.Ledger.Api.Era (
 ) where
 
 import Cardano.Ledger.Allegra (Allegra, AllegraEra)
+import Cardano.Ledger.Allegra.Scripts (translateTimelock)
+import Cardano.Ledger.Allegra.TxAuxData (AllegraTxAuxData (..))
+import Cardano.Ledger.Allegra.TxBody (AllegraEraTxBody (..), ValidityInterval (..))
+import qualified Cardano.Ledger.Allegra.TxBody as Allegra (TxBody (..))
 import Cardano.Ledger.Alonzo (Alonzo, AlonzoEra)
+import Cardano.Ledger.Alonzo.PParams (AlonzoPParams (appExtraEntropy), appD)
+import Cardano.Ledger.Alonzo.Scripts (AlonzoEraScript, upgradePlutusPurposeAsIx)
+import Cardano.Ledger.Alonzo.TxAuxData (AlonzoTxAuxData (..), AlonzoTxAuxDataRaw (..))
+import Cardano.Ledger.Alonzo.TxBody (AlonzoEraTxBody (..), TxBody (..))
+import Cardano.Ledger.Alonzo.TxWits (AlonzoTxWits (..), Redeemers (..), TxDats (..), unRedeemers)
 import Cardano.Ledger.Babbage (Babbage, BabbageEra)
+import Cardano.Ledger.Babbage.PParams (upgradeBabbagePParams)
+import Cardano.Ledger.Babbage.Tx
+import Cardano.Ledger.BaseTypes (StrictMaybe (..), isSJust)
+import Cardano.Ledger.Binary (mkSized, sizedValue)
+import Cardano.Ledger.Coin (Coin (..))
 import Cardano.Ledger.Conway (Conway, ConwayEra)
-import Cardano.Ledger.Core (
-  AtLeastEra,
-  AtMostEra,
-  ByronEra,
-  Era (..),
-  ExactEra,
-  ProtVerAtLeast,
-  ProtVerAtMost,
-  ProtVerInBounds,
-  atLeastEra,
-  atMostEra,
-  eraProtVerHigh,
-  eraProtVerLow,
- )
-import Cardano.Ledger.Mary (Mary, MaryEra)
+import Cardano.Ledger.Conway.Governance (VotingProcedures (..))
+import Cardano.Ledger.Conway.TxBody (TxBody (..))
+import Cardano.Ledger.Conway.TxCert (ConwayTxCertUpgradeError)
+import Cardano.Ledger.Core
+import Cardano.Ledger.Mary (Mary, MaryEra, TxBody (..))
+import Cardano.Ledger.Mary.TxBody (MaryEraTxBody (..))
+import Cardano.Ledger.MemoBytes (mkMemoizedEra)
+import Cardano.Ledger.Plutus.Data (upgradeData)
 import Cardano.Ledger.Shelley (Shelley, ShelleyEra)
+import Cardano.Ledger.Shelley.PParams
+import Cardano.Ledger.Shelley.Tx (ShelleyTx (..))
+import Cardano.Ledger.Shelley.TxAuxData (ShelleyTxAuxData (..))
+import Cardano.Ledger.Shelley.TxBody (ShelleyEraTxBody (..))
+import Cardano.Ledger.Shelley.TxWits (ShelleyTxWits (..))
+import Cardano.Ledger.Slot (SlotNo)
+import Control.Arrow (left)
+import Control.Monad (unless, when)
+import Data.Bifunctor (Bifunctor (first))
+import Data.Default (def)
+import Data.Kind (Type)
+import qualified Data.Map.Strict as Map
+import qualified Data.OSet.Strict as OSet
+import Data.Set (Set)
+import Data.Void (Void, absurd)
+import Lens.Micro
 
 -- | Sometimes it is useful to specify that a type corresponds to a latest era that is
 -- currently implemented
 type LatestKnownEra = ConwayEra
+
+class EraApi era where
```

Comment:
@lehins , your suggestion was: 
> (please also add possible constraints that are valid for all eras as superclasses

I suspect you mean `EraTx era, EraTxBody era..etc` 
But what benefits does it bring to add these superclasses here? 
I guess, in the test method, we would have fewer constraints to specify - is there more to it?